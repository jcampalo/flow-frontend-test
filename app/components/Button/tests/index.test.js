import React from 'react';
import { create } from 'react-test-renderer';

import Button from '../index';

const initial = {};
const renderComponent = props => {
  const finalProps = { ...initial, ...props };

  return (
    <Button {...finalProps}>
      Test
    </Button>
  );
};

describe('<Button />', () => {
  it('should render component', () => {
    const tree = create(renderComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component when is loading', () => {
    const props = {
      loading: true
    };
    const tree = create(renderComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
