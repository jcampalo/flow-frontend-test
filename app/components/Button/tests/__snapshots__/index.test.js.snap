// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render component 1`] = `
<div
  className="Wrapper__Wrapper-cZxMuO eDyakP"
>
  <button
    className="StyledButton__StyledButton-iQSVZX fjLvae"
    onClick={undefined}
  >
    Test
  </button>
</div>
`;

exports[`<Button /> should render component when is loading 1`] = `
<div
  className="Wrapper__Wrapper-cZxMuO eDyakP"
>
  <button
    className="StyledButton__StyledButton-iQSVZX fjLvae"
    onClick={undefined}
  >
    Test
  </button>
</div>
`;
